Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm

```diff
@@ -222,6 +222,46 @@
     completionHandler(std::nullopt);
 }
 
+void WebExtensionContext::declarativeNetRequestGetMatchedRules(std::optional<WebExtensionTabIdentifier> tabIdentifier, std::optional<double> minTimeStamp, CompletionHandler<void(std::optional<Vector<WebExtensionMatchedRuleParameters>> matchedRules, std::optional<String>)>&& completionHandler)
+{
+    RefPtr tab = tabIdentifier ? getTab(tabIdentifier.value()) : nil;
+
+    if (tabIdentifier && !tab) {
+        completionHandler(std::nullopt, toErrorString(@"declarativeNetRequest.getMatchedRules()", @"tabId", @"The specified tab does not exist."));
+        return;
+    }
+
+    if (!hasPermission(_WKWebExtensionPermissionDeclarativeNetRequestFeedback)) {
+        ASSERT(hasPermission(_WKWebExtensionPermissionActiveTab));
+
+        if (!hasPermission(_WKWebExtensionPermissionTabs, tab.get())) {
+            completionHandler(std::nullopt, toErrorString(@"declarativeNetRequest.getMatchedRules()", nil, @"The 'activeTab' permission has not been granted by the user for the specified tab."));
+            return;
+        }
+    }
+
+    static constexpr double oneSecondInMilliseconds = 1000;
+    NSDate *minTimeStampDate = minTimeStamp ? [NSDate dateWithTimeIntervalSince1970:minTimeStamp.value() / oneSecondInMilliseconds] : nil;
+
+    DeclarativeNetRequestMatchedRuleVector filteredMatchedRules;
+    for (auto matchedRule : matchedRules()) {
+        // FIXME: https://bugs.webkit.org/show_bug.cgi?id=262714 - we should be checking for permission if we don't have access to this URL.
+        if (!hasPermission(matchedRule.url))
+            continue;
```

Comment:
You are checking permission, but this FIXME is about _requesting_ permission.

Reply:
Yeah - I'll update the wording of the FIXME.